src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorType.cs

```diff
@@ -340,5 +352,132 @@ private Dictionary<string, FormulaValue> GetEnum()
 
             return enumDisplayNames.Zip(enumValues, (dn, ev) => new KeyValuePair<string, FormulaValue>(dn, ev)).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
         }
+
+        public bool Equals(ConnectorType other)
+        {
+            return Equals(this, other, 0);                  
+        }
+
+        internal static bool Equals(ConnectorType left, ConnectorType right, int depth)
+        {
+            if (left == null && right == null)
+            {
+                return true;
+            }
+
+            if (left == null || right == null)
+            {
+                return false;
+            }
+
+            return left.Name == right.Name &&
+                   left.DisplayName == right.DisplayName &&
+                   left.Description == right.Description &&
+                   left.IsRequired == right.IsRequired &&
+                   ArrayEquals(left.Fields, right.Fields, depth) &&
+                   ArrayEquals(left.HiddenFields, right.HiddenFields, depth) &&
+                   left.ExplicitInput == right.ExplicitInput &&
+                   left.IsEnum == right.IsEnum &&
+                   Enumerable.SequenceEqual((IList<FormulaValue>)left.EnumValues ?? Array.Empty<FormulaValue>(), (IList<FormulaValue>)right.EnumValues ?? Array.Empty<FormulaValue>()) &&
+                   Enumerable.SequenceEqual((IList<string>)left.EnumDisplayNames ?? Array.Empty<string>(), (IList<string>)right.EnumDisplayNames ?? Array.Empty<string>()) &&
+                   left.Visibility == right.Visibility &&
+                   ((left.Capabilities == null && right.Capabilities == null) || left.Capabilities.Equals(right.Capabilities)) &&
+                   left.KeyType == right.KeyType &&
+                   left.KeyOrder == right.KeyOrder &&
+                   left.Permission == right.Permission &&
+                   left.NotificationUrl == right.NotificationUrl &&
+                   ((left.HiddenRecordType == null && right.HiddenRecordType == null) || left.HiddenRecordType.Equals(right.HiddenRecordType)) &&
+                   left.Binary == right.Binary &&
+                   left.MediaKind == right.MediaKind &&
+                   Enumerable.SequenceEqual((IList<string>)left.ExternalTables ?? Array.Empty<string>(), (IList<string>)right.ExternalTables ?? Array.Empty<string>());
```

Comment:
We should also consider minimalizing it, maybe we should think of this comparison in terms of just coercion? Ie just compare field names... the problem is .Equals doesn't have good documentation on Aggregate type... but from usage it looks like its primarily used for coercion?

Reply:
Do we have some test, testing equality?